eth/ethtest/utils_test.go

```diff
@@ -0,0 +1,300 @@
+	logger *zap.Logger,
+	operator *testOperator,
+	ownerAddress *ethcommon.Address,
+	useMockCtrl bool,
+) (*eventhandler.EventHandler, *mocks.MockController, *gomock.Controller, operatorstorage.Storage, error) {
+	db, err := kv.NewInMemory(logger, basedb.Options{
+		Ctx: ctx,
+	})
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	storageMap := ibftstorage.NewStores()
+	nodeStorage, operatorData := setupOperatorStorage(logger, db, operator, ownerAddress)
+	testNetworkConfig := networkconfig.TestNetwork
+
+	keyManager, err := ekm.NewETHKeyManagerSigner(logger, db, testNetworkConfig, true, "")
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	ctrl := gomock.NewController(t)
+	bc := beacon.NewMockBeaconNode(ctrl)
+
+	contractFilterer, err := contract.NewContractFilterer(ethcommon.Address{}, nil)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	if useMockCtrl {
+		validatorCtrl := mocks.NewMockController(ctrl)
+
+		parser := eventparser.New(contractFilterer)
+
+		eh, err := eventhandler.New(
+			nodeStorage,
+			parser,
+			validatorCtrl,
+			testNetworkConfig.Domain,
+			validatorCtrl,
+			nodeStorage.GetPrivateKey,
+			keyManager,
+			bc,
+			storageMap,
+			eventhandler.WithFullNode(),
+			eventhandler.WithLogger(logger),
+		)
+
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+
+		validatorCtrl.EXPECT().GetOperatorData().Return(operatorData).AnyTimes()
+
+		return eh, validatorCtrl, ctrl, nodeStorage, nil
+	}
+
+	validatorCtrl := validator.NewController(logger, validator.ControllerOptions{
+		Context:         ctx,
+		DB:              db,
+		RegistryStorage: nodeStorage,
+		KeyManager:      keyManager,
+		StorageMap:      storageMap,
+		OperatorData:    operatorData,
+	})
+
+	parser := eventparser.New(contractFilterer)
+
+	eh, err := eventhandler.New(
+		nodeStorage,
+		parser,
+		validatorCtrl,
+		testNetworkConfig.Domain,
+		validatorCtrl,
+		nodeStorage.GetPrivateKey,
+		keyManager,
+		bc,
+		storageMap,
+		eventhandler.WithFullNode(),
+		eventhandler.WithLogger(logger),
+	)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	return eh, nil, ctrl, nodeStorage, nil
+}
+
+func setupOperatorStorage(
+	logger *zap.Logger,
+	db basedb.Database,
+	operator *testOperator,
+	ownerAddress *ethcommon.Address,
+) (operatorstorage.Storage, *registrystorage.OperatorData) {
+	if operator == nil {
+		logger.Fatal("empty test operator was passed", zap.Error(fmt.Errorf("empty test operator was passed")))
+	}
+
+	nodeStorage, err := operatorstorage.NewNodeStorage(logger, db)
+	if err != nil {
+		logger.Fatal("failed to create node storage", zap.Error(err))
+	}
+
+	operatorPubKey, err := nodeStorage.SetupPrivateKey(base64.StdEncoding.EncodeToString(operator.priv))
+	if err != nil {
+		logger.Fatal("could not setup operator private key", zap.Error(err))
+	}
+
+	_, found, err := nodeStorage.GetPrivateKey()
+	if err != nil || !found {
+		logger.Fatal("failed to get operator private key", zap.Error(err))
```

Comment:
if err is nil and not found, then error would be `failed to get operator private key error=nil` which looks confusing

Reply:
this is exact copy of what we got in `node.go`. [link](https://github.com/bloxapp/ssv/blob/stage/cli/operator/node.go#L433)

Reply:
I think it might be a good time to fix both 🙂